pkg/secretless/entrypoint/entrypoint.go

```diff
@@ -50,49 +53,77 @@ func StartSecretless(params *SecretlessOptions) {
 		log.Fatalln(err)
 	}
 
-	// Exit signal listener
+	// Optional Performance Profiling
+	handlePerformanceProfiling(params.ProfilingMode, exitListener)
 
-	// Coordinates processes interested in exit signals
-	exitListener := signal.NewExitListener()
+	configChangedChan, err := newConfigChangeChan(
+		params.ConfigFile,
+		params.ConfigManagerSpec,
+		params.FsWatchEnabled,
+	)
+	if err != nil {
+		log.Fatalln(err)
+	}
 
-	// Optional Performance Profiling
+	// Health check: Initialized
+	util.SetAppInitializedFlag()
+	util.SetAppIsLive(false)
 
-	handlePerformanceProfiling(params.ProfilingMode, exitListener)
+	var allServices internal.Service
+
+	configChangedFunc := func(cfg v2.Config) {
+		util.SetAppIsLive(false)
+		// Start Services
+		allServices = proxyservice.NewProxyServices(cfg, availPlugins, logger, evtNotifier)
+		err = allServices.Start()
+		if err != nil {
+			log.Fatalln(err)
+		}
 
-	// Start Services
+		// Health check: Live
+		util.SetAppIsLive(true)
+	}
 
-	allServices := proxyservice.NewProxyServices(cfg, availPlugins, logger, evtNotifier)
 	exitListener.AddHandler(func() {
-		fmt.Println("wait for all services signal")
+		fmt.Println("Received a stop signal")
+		if allServices == nil {
+			return
+		}
+
 		err := allServices.Stop()
 		if err != nil {
 			// Log but but allow cleanup of other subscribers to continue.
 			log.Println(err)
 		}
-	})
 
-	// Health check: Initialized
-	util.SetAppInitializedFlag()
-	util.SetAppIsLive(false)
+		os.Exit(0)
+	})
 
-	err = allServices.Start()
-	if err != nil {
-		log.Fatalln(err)
-	}
+	go func() {
+		for {
+			logger.Info("Waiting for configuration...")
+			cfg := <-configChangedChan
 
-	// Health check: Live
-	util.SetAppIsLive(true)
+			logger.Debug("Got new configuration")
+			configChangedFunc(cfg)
+		}
+	}()
```

Comment:
Looks like this goroutine is not stopped cleanly when we exit.  Not a blocker for me but should at least add a TODO.

Reply:
TODO added